robot/session_manager_test.go

```diff
@@ -25,7 +25,6 @@ func TestSessionManager(t *testing.T) {
 	}
 
 	sm := robot.NewSessionManager(r, config.DefaultSessionHeartbeatWindow)
-	defer sm.Close()
```

Comment:
If the asserts between here and line 53 fail, we won't close the session manager. If that only results in going from having 1 error to having 2 errors -- not a huge deal (unless the second error masks the first).

We have some options here (double closing a session manager I believe is actually fine -- certainly not true in general).

Reply:
good catch! added the deferred Close back and added a comment